### PR TITLE
Log to loggerfactory, support noop runs, healthcheck and monitor improvments

### LINKED ISSUE #7: Log out to standard Logger
A common pattern I end up using is to duplicate the messages I want to log to both the Logger and the TaskRepository:

```
String message = "doing something worth logging";
log.info(message);
context.log(message);
```

It would be nice to instead be able set a flag to send that output to both with a single call.

Not sure if this should be per task or global.

Possibly it would be enough to flip the log level on this line:

https://github.com/storebrand/scheduledtask/blob/main/scheduledtask-db-sql/src/main/java/com/storebrand/scheduledtask/db/sql/ScheduledTaskSqlRepository.java#L461

### LINKED ISSUE #9: Sort schedules by name on local introspection page
Schedules are not sorted on the local introspection page. They should be sorted by name, instead of current status where a map is just outputted as is.

### PR DESCRIPTION
**Pull request by DagBertelsen**
* Context.log/failed/done/dispached will now also output to the default logger. Resolves #7 
* All runs will append the schedule name to the MDC with the key "scheduledTask".
* Added support for NOOP runs. When seeing these runs in the monitorm and 
  if there is several consecutive runs in a row these will 
  be aggregated together and showing a count on how many where aggregated.
* NOOP runs has deleteNoopRunsAfterDays and keepMaxNoopRuns that can be set.
  They have defaults of deleteNoopRunsAfterDays = 7 and keepMaxNoopRuns=1440.
* Added checkbox to allow to inspect all NOOP runs, this only works if the 
  new integration is using the html function to render the HTML GUI.
* NOOP runs are shown with a color grey and a grey background.
* DONE runs are shown with a green color.
* FAILED runs are shown with red.
* The loglines are now shown under the row and by clicking this row they will appear.
* StackTraces are no longer shown in a modal box but rendered directly when the logs are shown.
* StackTraces will show the indents.
* BUG FIX: The HealthCheck "passed next run" where calculated from a memory variable on the given master.
  If there was a master switch this meant that this variable would be null or worse be days old and thus 
  giving a wrong or not working HealthCheck.
* HealthCheck changed from using true false to ✓/- when all is ok, and a describing word in caps when there 
  is an issue. This to assist in reading the healthCheck table.
* BUG FIX: In rare occations you could get a negative sleep number resulting in exception thrown. This would 
  cause the schedule to wait a bit before trying again and autorecover.
* If runOnce() is called inside a schedule run it will cause the schedule to retrigger immediately.
* Listing of schedules is now sorted by name. Resolves #9 
* HealthCheck table column "Thread running" is renamed to "Thread alive".

### CODE REVIEW

File: scheduledtask-testing/src/main/java/com/storebrand/scheduledtask/testing/MockRunContext.java [hunk 1]
```diff
@@ -165,6 +165,14 @@ public ScheduleStatus dispatched(String msg) {
         return createValidStatus();
     }
 
+
+    @Override
+    public ScheduleStatus noop(String msg) {
+        _state = State.NOOP;
+        _statusMessage = msg;
+        return createValidStatus();
```
Comment (khiimstb) [inline scheduledtask-testing/src/main/java/com/storebrand/scheduledtask/testing/MockRunContext.java:173]: missing log statement here

File: scheduledtask-annotation/src/main/java/com/storebrand/scheduledtask/annotation/ScheduledTask.java [hunk 2]
```diff
@@ -116,4 +124,11 @@
      * used to determine how many to keep.
      */
     int keepMaxFailedRuns() default 0;
+
+    /**
+     * Only keep this many noop runs. Older records will be deleted if there are more. This rule is set to 100 by
+     * default. If both this and {@link #keepMaxRuns()} is used then both will be applied, and the lowest number will be
+     * used to determine how many to keep.
+     */
+    int keepMaxNoopRuns() default ScheduledTaskBuilder.DEFAULT_KEEP_MAX_NOOP_RUNS;
```
Comment (khiimstb) [inline scheduledtask-annotation/src/main/java/com/storebrand/scheduledtask/annotation/ScheduledTask.java:133]: default for all other keep max is that they are disabled. We only enable deleteafterdays by default.

File: scheduledtask-core/src/main/java/com/storebrand/scheduledtask/ScheduledTaskConfig.java [hunk 3]
```diff
@@ -140,26 +152,40 @@ public int getDeleteSuccessfulRunsAfterDays() {
...
+                    || _keepMaxRuns > 0
+                    || _keepMaxFailedRuns > 0
+                    || _keepMaxNoopRuns > 0
+                    || _keepMaxSuccessfulRuns > 0;
         }
 
         public static class Builder {
             private int _keepMaxSuccessfulRuns;
             private int _keepMaxFailedRuns;
+            private int _keepMaxNoopRuns = ScheduledTaskBuilder.DEFAULT_KEEP_MAX_NOOP_RUNS; // Default 100
```
Comment (khiimstb) [inline scheduledtask-core/src/main/java/com/storebrand/scheduledtask/ScheduledTaskConfig.java:180]: I would just make it disabled by default

**Review by khiimstb — CHANGES_REQUESTED** [review]

**Comment by DagBertelsen** [comment]
Tested the new healthCheck:
image

Another screenshot showing that the Thread running is now renamed to Thread alive, it also shows when there is no issues with any of the schedules:
image

This change is tested in connect both with the support of the new html/outputStyleSheet/outputJAvaScript/json method calls and verified that the old way of initialising the HTML GUI also is supported. This where tested by deploying a custom version of MToolkitScheduledTask.

As a side note. The feature **Show all NOOPs** will only work in the new way of fetching the HTML GUI (by using the new html method). In the old way the page will refresh but the NOOPs will not show (only the aggregated ones).


Screenshot showing the default view of runs with fail, noops and done  
image

**Review by khiimstb — APPROVED** [review]

**Pull request merged**